src/App.tsx

```diff
@@ -1,10 +1,51 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { SortType } from './types/SortType';
+
+import { Todos } from './components/Todos/Todos';
+import { Footer } from './components/Footer/Footer';
+
+function getFilter(todos: Todo[], sortField: SortType) {
+  switch (sortField) {
+    case SortType.Active:
+      return todos.filter(todo => !todo.completed);
+    case SortType.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortField, setSortField] = useState(SortType.All);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const activeInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+
+        setTimeout(() => {
```

Comment:
[Not critical]
It also makes sense to clear this timeout when effect is cleared:
```
useEffect(() => {
  // set timeout
  return () => {
    // clear timeout
  };
}, []);
```